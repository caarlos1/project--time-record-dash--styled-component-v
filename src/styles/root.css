:root {
  --color-text: #2c3e50;
  --color-background: #ffffff;
  --color-primary: #605bff;
  --section-gap: 160px;
}

*,
*::before,
*::after {
  box-sizing: border-box;
  margin: 0;
  position: relative;
  font-weight: normal;
}

html {
  font-size: 62.5%;
  line-height: 62.5%;
}

body {
  min-height: 100vh;
  max-width: 100vw;
  color: var(--color-text);
  background: var(--color-background);
  transition: color 0.5s, background-color 0.5s;
  font-size: 1.6rem;
  line-height: 1.6rem;
  font-family: "Roboto", sans-serif;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  display: flex;
  position: relative;
  padding: 0 !important;
}

a {
  text-decoration: none;
  color: var(--color-text);
}

a:hover {
  color: var(--color-primary);
}
