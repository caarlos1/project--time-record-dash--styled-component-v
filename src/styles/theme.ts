export interface ThemeType {
  colors: {
    textColor: string;
    primaryColor: string;
  };
}

export const theme: ThemeType = {
  colors: {
    textColor: "#2c3e50",
    primaryColor: "#605bff",
  },
};

export const provide = { theme };
