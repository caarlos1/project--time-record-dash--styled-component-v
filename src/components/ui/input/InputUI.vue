<script setup lang="ts">
import * as Styled from "./InputUI.styles";
import { computed } from "vue";

const props = defineProps({
  modelValue: [String, Number],
  id: String,
  label: String,
  type: {
    type: String,
    default: "text",
  },
  placeholder: {
    type: String,
    default: "",
  },
});

const emit = defineEmits(["update:modelValue"]);
const inputValue = computed({
  get() {
    return props.modelValue;
  },
  set(value) {
    emit("update:modelValue", value);
  },
});
</script>

<template>
  <Styled.InputContainer>
    <Styled.Label v-if="props.label" :for="props.id">
      {{ props.label }}
    </Styled.Label>
    <Styled.Input v-bind="props" v-model="inputValue" />
  </Styled.InputContainer>
</template>
