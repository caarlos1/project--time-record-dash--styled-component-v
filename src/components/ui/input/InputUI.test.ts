import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";

import InputUI from "./InputUI.vue";

import { provide } from "@/styles/theme";
const mock = { id: 1, label: "Little Input" };

describe("InputUI", () => {
  it("should render a input with label", () => {
    const w = mount(InputUI, { props: { label: mock.label }, provide });
    expect(w.get("label").text()).toContain(mock.label);
  });

  it("should emit a event when change input value", () => {
    const w = mount(InputUI, { props: { label: mock.label }, provide });

    const input = w.get("input");
    const inputValue = "Input Value";
    input.setValue(inputValue);

    expect(input.element.value).toBe(inputValue);
    expect(w.emitted()).toHaveProperty("update:modelValue");
  });

  it("should match a snapshot", () => {
    const w = mount(InputUI, { props: { label: mock.label }, provide });
    expect(w.element).toMatchSnapshot();
  });
});
