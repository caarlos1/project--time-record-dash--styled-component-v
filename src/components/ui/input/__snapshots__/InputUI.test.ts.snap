// Vitest Snapshot v1

exports[`InputUI > should match a snapshot 1`] = `
<div
  class="lleXOl"
>
  <label
    class="lphNqp"
  >
    Little Input
  </label>
  <input
    class="hDLKVh"
    label="Little Input"
    placeholder=""
    type="text"
  />
</div>
`;
